sdk/identity/azure-identity/src/main/java/com/azure/identity/CredentialUnavailableException.java

```diff
@@ -5,21 +5,20 @@
 
 import com.azure.core.credential.AccessToken;
 import com.azure.core.credential.TokenCredential;
-import com.azure.core.exception.ClientAuthenticationException;
 
 /**
  * The exception thrown when a {@link TokenCredential} did not attempt to authenticate and retrieve {@link AccessToken},
  * as its prerequisite information or state was not available.
  */
-public class CredentialUnavailableException extends ClientAuthenticationException {
+public class CredentialUnavailableException extends RuntimeException {
```

Comment:
This should extend ClientAuthenticationException.

Reply:
If extend ClientAuthenticationException, There is a bug when running defaultazurecredential.
![image](https://user-images.githubusercontent.com/53427149/80553982-52ad2980-89fe-11ea-9bc4-2ac72a8c67bd.png)

Reply:
I am not able to reproduce this when used DefaultAzureCredential when no credential available with KeyVault Secrets client.
Are you running the above code of Storage with tracing enabled ?